src/utils/require-properties.ts

```diff
@@ -4,7 +4,7 @@ import * as Lodash from 'lodash';
 
 export function requireProperties(schema: TObjectJSONSchema, properties: string[], options: { preserveExisting?: boolean } = {}): TObjectJSONSchema {
   if (options.preserveExisting && schema.required) {
-    properties = Lodash.uniq(properties.concat(schema.required));
+    properties = Lodash.uniq(properties.concat(schema.required.map(key => key.toString())));
```

Comment:
Keys can now be symbols and the compiler was happy after adding `toString`. Unsure if this break some edge case 🤔

Reply:
Isn't the problem that the Object type allows for Symbol keys?

```typescript
//  // Seems reasonnable to force string keys on T?
export type TObjectJSONSchema<T extends { [key: string]: any; } = any> = TJSONSchema & { 
  maxProperties?: number;
  minProperties?: number;
  required?: (keyof T)[];
  additionalProperties?: boolean | TJSONSchema;
  definitions?: { [key: string]: TJSONSchema };
  properties?: { [key in keyof T]: TJSONSchema };
  patternProperties?: { [pattern: string]: TJSONSchema };
  dependencies?: { [key: string]: TJSONSchema | string[] };
}
```

Reply:
@SylvainEstevez btw, what about `unknown` in this case:

```Typescript
export type TObjectJSONSchema<T extends { [key: string]: unknown } = any> = TJSONSchema & {
```

Might not make a difference here in this case, but feels like the right choice.

Reply:
`unknown` is probably a better choice for libraries indeed 👍